hardhat.config.ts

```diff
@@ -30,7 +30,11 @@ if (PK) {
   };
 }
 
-if (["rinkeby", "mainnet"].includes(argv.network) && INFURA_KEY === undefined) {
+if (
+  ["rinkeby", "mainnet"].includes(argv.network) &&
+  NODE_URL === undefined &&
+  INFURA_KEY === undefined
```

Comment:
I'd prefer to remove the condition on `INFURA_KEY` because I always have it exported in my env, meaning that this check would always be false unless I unset it.
In general, I'd expect that `NODE_URL` has priority over `INFURA_KEY`, but feel free to leave it like this if you prefer.

Reply:
You are right, I read the condition incorrectly. :sweat_smile: